lib/govuk_personalisation/redirect.rb

```diff
@@ -0,0 +1,13 @@
+require "addressable"
+
+module GovukPersonalisation::Redirect
+  def redirect_with(base_url, additional_params = {})
+    return redirect_to base_url if additional_params.empty?
+
+    url = Addressable::URI.parse(base_url)
+    existing_params = url.query.split("&").map { |param| param.split("=") }
```

Comment:
I'm really unsure about parsing the URL like this, I still think string concatenation is better.

For example, this won't work for `https://www.gov.uk/search/all?keywords=foo=bar`, which is a totally valid URL which searches GOV.UK for `foo=bar`:

```
> url = Addressable::URI.parse("https://www.gov.uk/search/all?keywords=foo=bar")
=> #<Addressable::URI:0xf0c8 URI:https://www.gov.uk/search/all?keywords=foo=bar>

> existing_params = url.query.split("&").map { |param| param.split("=") }
=> [["keywords", "foo", "bar"]]

> url.query_values = existing_params
=> [["keywords", "foo", "bar"]]

> url.to_s
=> "https://www.gov.uk/search/all?keywords=foo"
```

Reply:
Ok that's a good challenge. I wasn't aware that `=` valid inside a query param value, i'd figured it was reserved like `?` or `&`
Without that I was reasonably sure what we'd done here was robust as we'd tested with keys only, keys and values and multiple and the null cases.

But you are right splitting on `=` doesn't work if it's in the value... 🤔 
@alex9smith lets drop the fanciness here and do string concatination then.

Means we can also drop addressable